app/controllers/users_controller.rb

```diff
@@ -209,6 +209,62 @@ def check_emails
     render json: failed_json, status: 403
   end
 
+  def update_primary_email
+    params.require(:email)
+
+    user = fetch_user_from_params
+    guardian.ensure_can_edit!(user)
```

Comment:
We have to use `UserGuardian#can_edit_email?` instead.

Reply:
I think this is still valid because we can't allow the primary email to change if `SiteSetting.email_editable` is false or if `SiteSetting.sso_overrides_email` is true